File: projects/order-creation-queries/core/src/main/java/com/robinfood/core/mappers/OrderDetailPaymentMethodMappers.java
package com.robinfood.core.mappers;

import com.robinfood.core.dtos.OrderDetailPaymentMethodDTO;
import com.robinfood.core.entities.OrderDetailPaymentMethodEntity;

public final class OrderDetailPaymentMethodMappers {

    private OrderDetailPaymentMethodMappers() {
        // this constructor is empty because it is a mapper class
    }

    public static OrderDetailPaymentMethodDTO toOrderDetailPaymentMethodDTO(
            OrderDetailPaymentMethodEntity orderDetailPaymentMethodEntity
    ) {
        return new OrderDetailPaymentMethodDTO(
                orderDetailPaymentMethodEntity.getDiscount(),
                orderDetailPaymentMethodEntity.getId(),
                orderDetailPaymentMethodEntity.getOriginId(),
                orderDetailPaymentMethodEntity.getSubtotal(),
                orderDetailPaymentMethodEntity.getTax(),
                orderDetailPaymentMethodEntity.getValue()
        );
    }
}

File: projects/posv2/core/src/main/kotlin/com/robinfood/core/mappers/OrderDetailPaymentMethodMappers.kt
package com.robinfood.core.mappers

import com.robinfood.core.constants.GlobalConstants.DEFAULT_LONG_VALUE
import com.robinfood.core.dtos.OrderDetailPaymentMethodDTO
import com.robinfood.core.entities.OrderDetailPaymentMethodEntity
import java.math.BigDecimal

fun OrderDetailPaymentMethodEntity.toOrderDetailPaymentMethodDTO(): OrderDetailPaymentMethodDTO? {
    return if (id == null) {
        null
    } else {
        return OrderDetailPaymentMethodDTO(
                discount = discount ?: BigDecimal.ZERO,
                id = id,
                originId = originId ?: DEFAULT_LONG_VALUE,
                subtotal = subtotal ?: BigDecimal.ZERO,
                tax = tax ?: BigDecimal.ZERO,
                value = value ?: BigDecimal.ZERO
        )
    }
}